File: 2024/01/a.mjs
import fs from "node:fs";

const compareNumbers = (a, b) => a - b;

function main(inputFile = "in") {
  const locations = { left: [], right: [] };
  fs.readFileSync(inputFile)
    .toString()
    .trim()
    .split("\n")
    .forEach((line) => {
      const [l, r] = line.split("   ").map(Number);
      locations.left.push(l);
      locations.right.push(r);
    });
  locations.left.sort(compareNumbers);
  locations.right.sort(compareNumbers);
  let result = 0;
  for (let i = 0; i < locations.left.length; i++) {
    const dist = Math.abs(locations.left[i] - locations.right[i]);
    result += dist;
  }
  console.log(result);
}

main();

File: 2024/01/src/a.rs
use std::fs;

pub fn main() -> i32 {
  let mut locations_left: Vec<i32> = Vec::new();
  let mut locations_right: Vec<i32> = Vec::new();
  fs::read_to_string("input/in").unwrap().trim().lines()
    .for_each(|line| {
      let mut split = line.split("   ");
      locations_left.push(split.next().unwrap().parse::<i32>().unwrap());
      locations_right.push(split.next().unwrap().parse::<i32>().unwrap());
    });
  locations_left.sort_unstable();
  locations_right.sort_unstable();
  let mut result = 0;
  for i in 0..locations_left.len() {
    let dist = (locations_left[i] - locations_right[i]).abs();
    result += dist;
  }
  result
}
